clang/lib/Sema/SemaChecking.cpp

```diff
@@ -1814,6 +1814,33 @@ static ExprResult PointerAuthAuthAndResign(Sema &S, CallExpr *Call) {
   return Call;
 }
 
+static ExprResult PointerAuthAuthLoadRelativeAndSign(Sema &S, CallExpr *Call) {
+  if (S.checkArgCount(Call, 6))
+    return ExprError();
+  if (checkPointerAuthEnabled(S, Call))
+    return ExprError();
+  const Expr *AddendExpr = Call->getArg(5);
+  std::optional<llvm::APSInt> AddendValue =
+      AddendExpr->getIntegerConstantExpr(S.Context);
+  const Expr *Arg = Call->getArg(5)->IgnoreParenImpCasts();
+  DeclRefExpr *DRE = cast<DeclRefExpr>(Call->getCallee()->IgnoreParenCasts());
+  FunctionDecl *FDecl = cast<FunctionDecl>(DRE->getDecl());
```

Comment:
Please put these inside the `if` block. Also, there's an `isIntegerConstantExpr` you can use if you don't care what the value is.